Mathlib/Data/Finset/Insert.lean

```diff
@@ -698,4 +698,25 @@ theorem pairwise_cons {a : α} (ha : a ∉ s) (r : α → α → Prop) :
 
 end Pairwise
 
+section Injective
+
+variable {f : α → β} {s : Finset α} {t : Finset β}
+
+variable (f s) in
+/-- A function `f` is injective on a finset `s` if and only if the image has no duplicates. -/
+theorem injOn_iff_nodup_map : Set.InjOn f s ↔ (Multiset.map f s.val).Nodup := by
+  induction s using cons_induction with
+  | empty => simp
+  | cons x xs hx h =>
+    simp [Set.injOn_insert (show x ∉ (xs : Set α) by simpa)]
```

Comment:
Can this become `simpa [...] using by tauto`? (Optional.)

Reply:
Even better, I found an existing mathlib lemma that proves this already!